validators/bls.go

```diff
@@ -0,0 +1,227 @@
+package validators
+
+import (
+	"bytes"
+	"encoding/hex"
+	"fmt"
+
+	"github.com/0xPolygon/polygon-edge/types"
+	"github.com/umbracle/fastrlp"
+)
+
+type BLSValidator struct {
+	Address      types.Address
+	BLSPublicKey []byte
+}
+
+func (v *BLSValidator) String() string {
+	return fmt.Sprintf(
+		"%s:%s",
+		v.Address.String(),
+		hex.EncodeToString(v.BLSPublicKey),
+	)
+}
+
+func (v *BLSValidator) Type() ValidatorType {
+	return BLSValidatorType
+}
+
+func (v *BLSValidator) Addr() types.Address {
+	return v.Address
+}
+
+func (v *BLSValidator) Copy() Validator {
+	pubkey := make([]byte, len(v.BLSPublicKey))
+	copy(pubkey, v.BLSPublicKey)
+
+	return &BLSValidator{
+		Address:      v.Address,
+		BLSPublicKey: pubkey,
+	}
+}
+
+func (v *BLSValidator) Equal(vr Validator) bool {
+	vv, ok := vr.(*BLSValidator)
+	if !ok {
+		return false
+	}
+
+	return v.Address == vv.Address && bytes.Equal(v.BLSPublicKey, vv.BLSPublicKey)
+}
+
+func (v *BLSValidator) MarshalRLPWith(arena *fastrlp.Arena) *fastrlp.Value {
+	vv := arena.NewArray()
+
+	vv.Set(arena.NewBytes(v.Address[:]))
+	vv.Set(arena.NewBytes(v.BLSPublicKey))
+
+	return vv
+}
+
+func (v *BLSValidator) UnmarshalRLPFrom(p *fastrlp.Parser, val *fastrlp.Value) error {
+	elems, err := val.GetElems()
+	if err != nil {
+		return err
+	}
+
+	if len(elems) < 2 {
+		return fmt.Errorf("incorrect number of elements to decode BLSValidator, expected 2 but found %d", len(elems))
+	}
+
+	if err := elems[0].GetAddr(v.Address[:]); err != nil {
+		return fmt.Errorf("failed to decode Address: %w", err)
+	}
+
+	if v.BLSPublicKey, err = elems[1].GetBytes(v.BLSPublicKey); err != nil {
+		return fmt.Errorf("failed to decode BLSPublicKey: %w", err)
+	}
+
+	return nil
+}
```

Comment:
Why do you need RLP encoding?

Reply:
Validators is put in IBFT Extra and Validator may be set in Header.